scripts/speech_recognition/partial_convertion_to_tarred_audio_dataset.py

```diff
@@ -0,0 +1,199 @@
+- Make sure the paths to the manifest and metadata files are correct and accessible.
+
+## Usage
+
+### Script Execution
+
+To run the script, use the following command:
+
+python partial_convertion_to_tarred_audio_dataset.py \
+    # the path to the tarred manifest file that contains the entries for the shards you want to process. This option is mandatory.
+    --tarred_manifest_filepath=<path to the tarred manifest file > \
+    # any other optional argument
+    --output_dir=<output directory for tarred shards> \
+    --shards_to_tar=<shard IDs to be tarred> \
+    --num_workers=-1 \
+    --dataset_metadata_filepath=<dataset metadata YAML filepath>
+
+Example:
+python partial_convertion_to_tarred_audio_dataset.py \
+    tarred_manifest_filepath="path/to/manifest.json" \
+    shards_to_tar="0:3"
+"""
+
+
+def select_shards(manifest_filepath: str, shards_to_tar: str, slice_with_offset: bool = False):
+    """
+    Selects and returns a subset of shards from the tarred manifest file.
+
+    Args:
+        manifest_filepath (str): The path to the tarred manifest file.
+        shards_to_tar (str): A range or list of shard IDs to select, e.g., "0:5" or "0,1,2".
+        slice_with_offset (bool, optional): If True, slices entries based on audio offsets. Defaults to False.
+
+    Raises:
+        FileNotFoundError: If the manifest file does not exist.
+        KeyError: If `slice_with_offset` is enabled but required fields are missing in the manifest entries.
+
+    Returns:
+        Dict[int, List[Dict[str, any]]]: A dictionary where the keys are shard IDs and the values are lists of entries for those shards.
+    """
+    shard_ids = []
+    if shards_to_tar != "all":
+        if ":" not in shards_to_tar:
+            shard_ids = [int(shards_to_tar)]
+        else:
+            start_shard_idx, end_shard_idx = map(
+                lambda x: int(x.strip()) if x.strip() else None, shards_to_tar.split(":")
+            )
+            shard_ids = list(range(start_shard_idx, end_shard_idx))
+
+    entries_to_shard = {}
+    with open(manifest_filepath, 'r') as manifest:
+        for line in tqdm(manifest, desc="Selecting shards"):
+            entry = json.loads(line)
+            if shards_to_tar == "all" or entry['shard_id'] in shard_ids:
+                if entry['shard_id'] not in entries_to_shard:
+                    entries_to_shard[entry['shard_id']] = []
+
+                if slice_with_offset:
+                    if 'abs_audio_filepath' not in entry or 'source_audio_offset' not in entry:
+                        raise KeyError(
+                            f"`slice_with_offset` is enabled, but `abs_audio_filepath` and/or `source_audio_offset` are not found in the entry:\n{entry}."
+                        )
+                    entry['audio_filepath'] = entry['abs_audio_filepath']
+                    entry['offset'] = entry['source_audio_offset']
```

Comment:
should we delete `abs_audio_filepath` and `source_audio_offset` keys from the entry, as they are no longer useful at this point?